src/app/AiPlayer.ts

```diff
@@ -0,0 +1,195 @@
+import { AnimalNames } from '~src/Enums/AnimalNamesEnum';
+import { AnimalRoles } from '~src/Enums/AnimalRolesEnum';
+import { Player } from '../Player';
+import { GameController } from './GameController';
+import { Offer } from './Trade';
+import { ConvertAnimalName } from './utils/ConvertAnimalName';
+
+export class AiPlayer extends Player {
+  /**
+   * Makes a move for AI Player
+   * @param gameController for the game that the player is participating in
+   */
+  makeAMove(gameController: GameController): void {
+    this.trade(gameController);
+    if (gameController.theGameProcessor.checkWin()) {
+      gameController.checkIfGameIsWon();
+      return;
+    }
+    gameController.breed();
+    if (gameController.theGameProcessor.checkWin()) {
+      gameController.checkIfGameIsWon();
+      return;
+    }
+  }
+  private trade(gameController: GameController): void {
```

Comment:
This method is really long. Could there be a way to split it into smaller chunks?

Reply:
For now, it'd be very hard, because of the return statements that are used in this method.